v5/core/utils/periodic_task.go

```diff
@@ -1,15 +1,24 @@
 package utils
 
-import "time"
+import (
+	"time"
+)
 
 func RunPeriodicTask(action func(), period time.Duration, quit <-chan struct{}) {
-	ticker := time.NewTicker(period)
-	defer ticker.Stop()
+	go func() {
+		// Start a new ticker to execute the task
+		ticker := time.NewTicker(period)
 
-	select {
-	case <-ticker.C:
-		action()
-	case <-quit:
-		return
-	}
+		// Loop forever, executing the task on each tick. Only stop when the "quit"
+		// channel tells us to, or the channel is closed.
+		for {
+			select {
+			case <-ticker.C:
+				action()
+			case <-quit:
+				ticker.Stop()
+				return
+			}
+		}
```

Comment:
This is the meat of the fix.  The original `select` only waited to check the 1st tick and then exited.  This fix obviously keeps checking in a `for` loop until told to stop by the `quit` channel.

Reply:
can add a test if you think it will be fun (check it is called at least twice?)
seems good to me without test anynway

Reply:
Added `v5/core/utils/periodic_task_test.go`.  Tried to add a test to `core_test.go` to verify that a `ConfigurationFetchedHandler` was called multiple times after `rox.Setup`, but the code is too tightly coupled to be tested and didn't want to take time/risk to change it at this point.

Reply:
Yes, hard to test core, too many dependencies 👍